addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/converter/ZWaveAlarmConverter.java

```diff
@@ -82,27 +83,92 @@ public ZWaveAlarmConverter() {
+        }
+        return null;
+    }
+
+    private State handleAlarmReport(ZWaveThingChannel channel, ZWaveAlarmValueEvent eventAlarm) {
+        String alarmType = channel.getArguments().get("type");
+        // Don't trigger event if this item is bound to another alarm type
+        if (alarmType != null) {
+            // alarmType should be a number
+            Integer alarmTypeInt = null;
+            try {
+                alarmTypeInt = Integer.valueOf(alarmType);
+                if (alarmTypeInt != eventAlarm.getAlarmType()) {
+                    return null;
+                }
+            } catch (NumberFormatException e) {
+                // try to interpret it as a enum value
+                if (AlarmType.valueOf(alarmType).getKey() != eventAlarm.getAlarmType()) {
+                    return null;
+                }
+            }
+        }
+
+        State state = null;
+        switch (channel.getDataType()) {
+            case OnOffType:
+                state = eventAlarm.getValue() == 0 ? OnOffType.OFF : OnOffType.ON;
+                break;
+            case OpenClosedType:
+                state = eventAlarm.getValue() == 0 ? OpenClosedType.CLOSED : OpenClosedType.OPEN;
+                break;
+            default:
+                logger.warn("No conversion in {} to {}", this.getClass().getSimpleName(), channel.getDataType());
+                break;
+        }
+        return state;
+
+    }
+
+    private State handleNotifictionReport(ZWaveThingChannel channel, ZWaveAlarmValueEvent eventAlarm) {
         String alarmType = channel.getArguments().get("type");
         String alarmEvent = channel.getArguments().get("event");
-        ZWaveAlarmValueEvent eventAlarm = (ZWaveAlarmValueEvent) event;
 
         // Don't trigger event if this item is bound to another alarm type
-        if (alarmType != null && AlarmType.valueOf(alarmType) != eventAlarm.getAlarmType()) {
+        if (alarmType != null && AlarmType.valueOf(alarmType) != eventAlarm.getZwaveAlarmType()) {
             return null;
         }
 
-        // Check the event type
+        // Check the event type( this will only report the alarm activation. Not deactivation.
         if (alarmEvent != null && Integer.parseInt(alarmEvent) != eventAlarm.getAlarmEvent()) {
             return null;
         }
 
+        // handle event 0xfe as 'clear the event'
+        int event = eventAlarm.getAlarmEvent() == 0xfe ? 0 : eventAlarm.getAlarmEvent();
+
         State state = null;
         switch (channel.getDataType()) {
             case OnOffType:
-                state = (Integer) event.getValue() == 0 ? OnOffType.OFF : OnOffType.ON;
+                state = event == 0 ? OnOffType.OFF : OnOffType.ON;
                 break;
             case OpenClosedType:
-                state = (Integer) event.getValue() == 0 ? OpenClosedType.CLOSED : OpenClosedType.OPEN;
+                if (AlarmType.ACCESS_CONTROL.equals(eventAlarm.getZwaveAlarmType())) {
```

Comment:
Do you think that this can be 'table' driven rather than a bunch of cascading case statements - it would be more maintainable I think than lots of case statements....